Catkin build fails with multiple packages and clean workspace

To reproduce:
cd to your catkin workspace
catkin clean then catkin build.
This error will appear
fatal error: ximea_ros_cam/XiGetImage.h: No such file or directory
compilation terminated.
make[2]: *** [CMakeFiles/ximea_ros_cam.dir/src/ximea_ros_cam.cpp.o] Error 1
make[1]: *** [CMakeFiles/ximea_ros_cam.dir/all] Error 2
make: *** [all] Error 2

Cause:
The XiGetImage.h should be created from the XiGetImage.msg file but does not exist yet.

Solution:
Add the messages generated by the package as a dependency to the package.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -66,6 +66,9 @@ target_link_libraries(${PROJECT_NAME}
     yaml-cpp
 )
 
+## Package depends on its generated messages
+add_dependencies(${PROJECT_NAME} ${PROJECT_NAME}_generate_messages_cpp)
+
 ## Declare a C++ executable
 add_executable(${PROJECT_NAME}_node src/ximea_ros_cam_node.cpp)
 

